### DISCUSSION TITLE
NetBSD/OpenBSD: find(1) does not support -iwholename

### DISCUSSION THREAD

**Discussion by 0-wiz-0** [minad/consult#989] [Q&A]
https://github.com/minad/consult/blob/7533460e530a1e7abc89b4da8ad99b696fca8a30/consult.el#L282
uses find(1) with `-iwholename`, which is not supported by the find(1) implementations on NetBSD and OpenBSD.
OpenBSD has `-iname`; NetBSD has `-iname` and `-iregex`.
Do you have an idea how to support them better?
Thank you!

**Comment by minad** [comment]
The GNU find man page also mentions -ipath. Would that work on the BSDs? Besides that we could try to detect the find version and act accordingly. However the requirement is that the BSD find must support whole path matching in some way. If this is not possible, we cannot support them. In that case, I suggest that you install the GNU utils instead.

**Reply by minad** [reply]
I looked at the [OpenBSD find man page](https://man.openbsd.org/find.1) and it seems we cannot easily support it. The iwholename problem is easily fixable by changing consult-find-args, but the consult--find-make-builder also uses the unsupported -iregex (and not globbing).

**Comment by 0-wiz-0** [comment]
Correct me if I'm wrong, but:
according to the [FreeBSD man page](https://man.freebsd.org/cgi/man.cgi?query=find&apropos=0&sektion=0&manpath=FreeBSD+14.0-RELEASE+and+Ports&arch=default&format=html), `-iwholename` is the same as `-ipath`, and `-ipath` is the case-insensitive match version of `-path`. OpenBSD and NetBSD do support `-path`, could we use
```
 "find . -not ( -path */.[A-Za-z]* -prune )" 
```
i.e. adapt the pattern to include the upper-case versions?

**Reply by minad** [reply]
The FreeBSD man page is not helpful here, since FreeBSD supports all the GNU find options. The limited find variants of NetBSD and OpenBSD are the problem.

**Reply by 0-wiz-0** [reply]
Sorry, but that didn't address my proposal at all.
If we change the pattern to include A-Z too, then the lower- and upper-case versions of a-z are in the pattern and we don't need to use the case-insensitive variant of the command line option.
And the case-sensitive variant of the option is support on all three BSDs.

**Reply by minad** [reply]
> If we change the pattern to include A-Z too, then the lower- and upper-case versions of a-z are in the pattern and we don't need to use the case-insensitive variant of the command line option.

That's right. I think your proposal will fix NetBSD. (FreeBSD is supported anyway as I wrote above.)

(EDIT: See commit https://github.com/minad/consult/commit/c87b0bf06de0c3cb60bc8d257c770cb981ddcd19)

> And the case-sensitive variant of the option is support on all three BSDs.

But -iregex is not supported by OpenBSD, and this is needed by the find command line builder function.

**Reply by 0-wiz-0** [reply]
Thank you for the commit!
